src/test/java/com/synopsys/integration/jenkins/extensions/JenkinsIntLoggerTest.java

```diff
@@ -0,0 +1,197 @@
+            Arguments.of("", null, LogLevel.TRACE, "\n"),
+            Arguments.of(textMessage, null, LogLevel.OFF, expectedText),
+            Arguments.of(textMessage, null, LogLevel.ERROR, expectedText),
+            Arguments.of(textMessage, null, LogLevel.WARN, expectedText),
+            Arguments.of(textMessage, null, LogLevel.INFO, expectedText),
+            Arguments.of(textMessage, null, LogLevel.DEBUG, expectedText),
+            Arguments.of(textMessage, null, LogLevel.TRACE, expectedText)
+        );
+    }
+
+    private static Stream<Arguments> exceptionTestData() {
+        String exceptionMessage = "This is a test exception";
+        RuntimeException runtimeException = new RuntimeException(exceptionMessage);
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter printWriter = new PrintWriter(stringWriter);
+        runtimeException.printStackTrace(printWriter);
+        String expectedException = stringWriter.toString() + "\n";
+
+        return Stream.of(
+            Arguments.of(null, runtimeException, LogLevel.OFF, expectedText),
+            Arguments.of(null, runtimeException, LogLevel.ERROR, expectedException),
+            Arguments.of(null, runtimeException, LogLevel.WARN, expectedException),
+            Arguments.of(null, runtimeException, LogLevel.INFO, expectedException),
+            Arguments.of(null, runtimeException, LogLevel.DEBUG, expectedException),
+            Arguments.of(null, runtimeException, LogLevel.TRACE, expectedException),
+            Arguments.of(textMessage, runtimeException, LogLevel.OFF, expectedText + expectedException),
+            Arguments.of(textMessage, runtimeException, LogLevel.ERROR, expectedText + expectedException),
+            Arguments.of(textMessage, runtimeException, LogLevel.WARN, expectedText + expectedException),
+            Arguments.of(textMessage, runtimeException, LogLevel.INFO, expectedText + expectedException),
+            Arguments.of(textMessage, runtimeException, LogLevel.DEBUG, expectedText + expectedException),
+            Arguments.of(textMessage, runtimeException, LogLevel.TRACE, expectedText + expectedException)
+        );
+    }
+
+    @Test
+    public void testJenkinsIntLoggerTaskListener() {
+        TaskListener taskListener = Mockito.mock(TaskListener.class);
+        Mockito.when(taskListener.getLogger()).thenReturn(new PrintStream(byteArrayOutputStream));
+
+        JenkinsIntLogger jenkinsIntLogger = new JenkinsIntLogger(taskListener);
+        jenkinsIntLogger.setLogLevel(LogLevel.OFF);
+        jenkinsIntLogger.alwaysLog(textMessage);
+
+        assertEquals(taskListener, jenkinsIntLogger.getTaskListener());
+        assertEquals(expectedText, byteArrayOutputStream.toString());
+    }
+
+    @ParameterizedTest
+    @MethodSource("txtTestData")
+    public void testJenkinsIntLogger_alwaysLog(String txt, Throwable e, LogLevel logLevel, String expected) {
+        if (e == null) {
```

Comment:
It appears that the intention of the test is for e to always be null. To that end, I'd prefer the test to fail if e is not null than to quietly succeed while doing nothing!

Not passing e at all is definitely superior to that, but I see why you're doing it.